adapters/adf/adf.go

```diff
@@ -47,6 +52,27 @@ func (a *adapter) MakeRequests(request *openrtb2.BidRequest, requestInfo *adapte
 
 		imp.TagID = adfImpExt.MasterTagID.String()
 		validImps = append(validImps, imp)
+
+		if adfImpExt.PriceType != "" {
+			var requestExt adfRequestExt
+
+			if len(request.Ext) > 0 {
+				if err := json.Unmarshal(request.Ext, &requestExt); err != nil {
+					errors = append(errors, err)
+					continue
+				}
+			} else {
+				requestExt = adfRequestExt{}
+			}
+
+			requestExt.PriceType = adfImpExt.PriceType
+
+			var err error
+			if request.Ext, err = json.Marshal(&requestExt); err != nil {
```

Comment:
Line 60: you marshal req.Ext as many times as namy imps are thre in reqiest - it's located inside `for _, imp := range request.Imp {`. This should be moved ouside the for loop. 
Line 71 sets new request.ext on every iteration over imp array. It `means req.ext.PriceType` will have the value from last imp in loop. 
How will it work on your saide if there are two or more imp in request with different `PriceType`?

Reply:
All priceTypes need to be the same, but really only the first one should be used and not the last one if they should differ.
I took a bad shortcut there. Me programming Go is also a bit like https://youtu.be/C1Sw0PDgHU4.